server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -340,6 +346,13 @@ private void tryAllHandlers(final RestRequest request, final RestChannel channel
         handleBadRequest(uri, requestMethod, channel);
     }
 
+    private void handleCompatibleNotAllowed(String rawPath, RestChannel channel) throws IOException {
+        String msg = "Compatible api can be only accessed with Compatible Header. Path used: " + rawPath;
```

Comment:
I am on the fence about this message, but leaning towards removing it.  I am not sure we need to tell people that this could be a compatible path.

Reply:
I guess for future clients versions there would always be passed a header with a version. But in transition from 7 to 8, I guess some people would stay with old (like 7.0) client and won't have a header? This way they could find out quickly about the compatible API.

I added it mostly because I found it useful when developing - but happy to remove it too.

Reply:
Let's remove for now. We can add it later if need be.

Reply:
will do. btw we are at the moment returning 400 bad request when handler was not found.